dev/benchmarks/macrobenchmarks/lib/src/web/recorder.dart

```diff
@@ -503,54 +490,186 @@ class _WidgetBuildRecorderHostState extends State<_WidgetBuildRecorderHost> {
 class Timeseries {
   Timeseries(this.name);
 
+  /// The label of this timeseries used for debugging and result inspection.
   final String name;
 
   /// List of all the values that have been recorded.
   ///
   /// This list has no limit.
-  final List<num> _allValues = <num>[];
-
-  /// List of values that are being used for measurement purposes.
-  ///
-  /// [average], [standardDeviation] and [noise] are all based on this list, not
-  /// the [_allValues] list.
-  final List<num> _measuredValues = <num>[];
+  final List<double> _allValues = <double>[];
 
   /// The total amount of data collected, including ones that were dropped
   /// because of the sample size limit.
   int get count => _allValues.length;
 
-  /// Computes the average value of the measured values.
-  double get average => _computeAverage(name, _measuredValues);
-
-  /// Computes the standard deviation of the measured values.
-  double get standardDeviation =>
-      _computeStandardDeviationForPopulation(name, _measuredValues);
-
-  /// Computes noise as a multiple of the [average] value.
-  ///
-  /// This value can be multiplied by 100.0 to get noise as a percentage of
-  /// the average.
+  /// Extracts useful statistics out of this timeseries.
   ///
-  /// If [average] is zero, treats the result as perfect score, returns zero.
-  double get noise => average > 0.0 ? standardDeviation / average : 0.0;
+  /// See [TimeseriesStats] for more details.
+  TimeseriesStats computeStats() {
+    // The first few values we simply discard and never look at. There's from warm-up phase.
```

Comment:
Maybe you meant "They're from warm-up phase"?

Reply:
Done.